test/metabase/test/data/mbql_query_impl.clj

```diff
@@ -8,7 +8,18 @@
 
 (set! *warn-on-reflection* true)
 
-(def ^:dynamic *id-fn-symb*              'metabase.test.data/id)
+#_:clj-kondo/ignore
+(defn druid-id-fn
+  "I have a strong feeling this should be handled differently! Let's discuss that during review!"
```

Comment:
I agree, this is pretty icky. I think a wrapper macro that binds `*id-fn-symb*` to something different in the places you need it (`timeseries-query-processor-test`) is a better solution

Reply:
I like that idea. However I've yet found no way how to make that work. See [this comment](https://github.com/metabase/metabase/pull/40293#discussion_r1589446353).